gumps/Newfile_gump.cc

```diff
@@ -919,7 +918,7 @@ void Newfile_gump::LoadSaveGameDetails() {
 			npc = gwin->get_npc(partyman->get_member(i - 1));
 
 		std::string namestr = npc->get_npc_name();
-		strncpy(cur_party[i].name, namestr.c_str(), 18);
+		memcpy(cur_party[i].name, namestr.c_str(), 18);
```

Comment:
More undefined behavior here whenever `namestr.size() < 17`. You need to call `namestr.resize(18, '\0')` before this line.